obal/tito_extensions/git_annex_spec_builder.py

```diff
@@ -0,0 +1,153 @@
+import os
+import shutil
+from distutils.version import LooseVersion as loose_version
+from pkg_resources import require
+
+from tito.compat import getstatusoutput
+from tito.config_object import ConfigObject
+from tito.builder import GitAnnexBuilder
+from tito.builder.main import BuilderBase
+from tito.common import error_out, debug, run_command, get_spec_version_and_release, \
+    find_spec_like_file, get_relative_project_dir_cwd, warn_out
+
+
+class GitAnnexSpecBuilder(GitAnnexBuilder):
+
+    def __init__(self, name=None, tag=None, build_dir=None,
+                 config=None, user_config=None,
+                 args=None, **kwargs):
+
+        """
+        name - Package name that is being built.
+
+        version - Version and release being built.
+
+        tag - The git tag being built.
+
+        build_dir - Temporary build directory where we can safely work.
+
+        config - Merged configuration. (global plus package specific)
+
+        user_config - User configuration from ~/.titorc.
+
+        args - Optional arguments specific to each builder. Can be passed
+        in explicitly by user on the CLI, or via a release target config
+        entry. Only for things which vary on invocations of the builder,
+        avoid using these if possible.  *Given in the format of a dictionary
+        of lists.*
+        """
+        ConfigObject.__init__(self, config=config)
+        BuilderBase.__init__(self, name=name, build_dir=build_dir, config=config,
+                             user_config=user_config, args=args, **kwargs)
+        self.build_tag = tag
+
+        self.build_version = self._get_build_version()
+        self.git_commit_id = self.build_version
+
+        if kwargs and 'options' in kwargs:
+            warn_out("'options' no longer a supported builder constructor argument.")
+
+        if self.config.has_option("requirements", "tito"):
+            if loose_version(self.config.get("requirements", "tito")) > \
+                    loose_version(require('tito')[0].version):
+                error_out([
+                    "tito version %s or later is needed to build this project." %
+                    self.config.get("requirements", "tito"),
+                    "Your version: %s" % require('tito')[0].version
+                ])
+
+        self.display_version = self._get_display_version()
+
+        self.relative_project_dir = get_relative_project_dir_cwd(self.git_root)
+
+        tgz_base = self._get_tgz_name_and_ver()
+        self.tgz_filename = tgz_base + ".tar.gz"
```

Comment:
Would this fail on [nodejs-babel-cli](https://github.com/theforeman/foreman-packaging/tree/8446d10183de7af1bf73ca100ff181684f3b48fd/nodejs-babel-cli) since the extension is .tgz? It also has multiple source file.

Reply:
Good question, but this tarball is actually generated by tito, it wraps up the spec and sources into a tarball for it's inner workings so this shouldn't break any one package.